### PR TITLE
vendor runc v1.0.0-rc91

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Required by https://github.com/containerd/containerd/pull/4357

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
```
--- FAIL: TestPrivilegedDevices (0.04s)
##[error]    container_create_unix_test.go:1185: TestCase "expect no host devices when privileged is false"
##[error]    container_create_unix_test.go:1185: TestCase "expect no host devices when privileged is false and privilegedWithoutHostDevices is true"
##[error]    container_create_unix_test.go:1185: TestCase "expect host devices when privileged is true"
```
https://github.com/containerd/cri/pull/1521/checks?check_run_id=826161016

**Comment by AkihiroSuda** [comment]
The failure is because `libcontainer/devices.HostDevices()` returns 416 devices on rc91, 187 devices on rc10.

https://github.com/containerd/cri/blob/8e0b789c9ade07497283546c0bde0fb5f99310cb/pkg/server/container_create_unix_test.go#L1196-L1197

https://github.com/opencontainers/runc/blob/v1.0.0-rc10/libcontainer/devices/devices.go#L64-L67
https://github.com/opencontainers/runc/blob/v1.0.0-rc91/libcontainer/devices/devices.go#L64-L67

cc @cyphar @kolyshkin

**Comment by AkihiroSuda** [comment]
```go
package main

import (
        "fmt"

        "github.com/opencontainers/runc/libcontainer/devices"
)

func main() {
        devs, err := devices.HostDevices()
        if err != nil {
                panic(err)
        }
        for _, dev := range devs {
                fmt.Println(dev.Path)
        }
}
```

```diff
--- rc90-sorted 2020-07-02 15:18:15.816017724 +0900
+++ rc91-sorted 2020-07-02 15:18:20.436162364 +0900
@@ -1,18 +1,217 @@
 /dev/agpgart
 /dev/autofs
+/dev/block/11:0
+/dev/block/7:0
+/dev/block/7:1
+/dev/block/7:2
+/dev/block/7:3
+/dev/block/7:4
+/dev/block/7:5
+/dev/block/7:6
+/dev/block/7:7
+/dev/block/8:0
+/dev/block/8:1
+/dev/block/8:2
 /dev/bsg/3:0:0:0
 /dev/bsg/32:0:0:0
 /dev/btrfs-control
+/dev/cdrom
+/dev/cdrw
+/dev/char/10:1
+/dev/char/10:175
+/dev/char/10:183
+/dev/char/10:196
+/dev/char/10:200
+/dev/char/10:223
...
+/dev/dri/by-path/pci-0000:00:0f.0-card
+/dev/dri/by-path/pci-0000:00:0f.0-render
 /dev/dri/card0
 /dev/dri/renderD128
+/dev/dvd
 /dev/ecryptfs
 /dev/fb0
+/dev/fd
 /dev/full
 /dev/fuse
 /dev/hpet
 /dev/hwrng
+/dev/initctl
+/dev/input/by-path/platform-i8042-serio-0-event-kbd
+/dev/input/by-path/platform-i8042-serio-1-event-mouse
+/dev/input/by-path/platform-i8042-serio-1-mouse
 /dev/input/event0
 /dev/input/event1
 /dev/input/event2
@@ -23,6 +222,7 @@
 /dev/kmsg
 /dev/kvm
 /dev/lightnvm/control
+/dev/log
 /dev/loop0
 /dev/loop1
 /dev/loop2
@@ -44,6 +244,7 @@
 /dev/ptmx
 /dev/random
 /dev/rfkill
+/dev/rtc
 /dev/rtc0
 /dev/sda
 /dev/sda1
@@ -54,6 +255,9 @@
 /dev/snd/seq
 /dev/snd/timer
 /dev/sr0
+/dev/stderr
+/dev/stdin
+/dev/stdout
 /dev/tty
 /dev/tty0
 /dev/tty1
```

File: vendor.conf [hunk 1]
```diff
@@ -36,28 +36,30 @@ github.com/Microsoft/go-winio                       v0.4.14
 github.com/Microsoft/hcsshim                        v0.8.9
 github.com/opencontainers/go-digest                 v1.0.0
 github.com/opencontainers/image-spec                v1.0.1
-github.com/opencontainers/runc                      v1.0.0-rc10
-github.com/opencontainers/runtime-spec              v1.0.2
+github.com/opencontainers/runc                      v1.0.0-rc91
+github.com/opencontainers/runtime-spec              237cc4f519e2e8f9b235bacccfa8ef5a84df2875
```
Comment (mikebrow) [inline vendor.conf:40]: kk pickup for errno fix.

File: vendor.conf [hunk 2]
```diff
@@ -16,14 +16,14 @@ github.com/containerd/go-runc                       7016d3ce2328dd2cb1192b2076eb
...
-github.com/cpuguy83/go-md2man                       v1.0.10
+github.com/cpuguy83/go-md2man/v2                    v2.0.0
 github.com/docker/go-events                         e31b211e4f1cd09aa76fe4ac244571fab96ae47f
 github.com/docker/go-metrics                        v0.0.1
 github.com/docker/go-units                          v0.4.0
 github.com/godbus/dbus/v5                           v5.0.3
 github.com/gogo/googleapis                          v1.3.2
 github.com/gogo/protobuf                            v1.3.1
-github.com/golang/protobuf                          v1.3.3
+github.com/golang/protobuf                          v1.3.5
```
Comment (mikebrow) [inline vendor.conf:26]: kk i see these are also being pushed to containerd/containerd.. in parallel..

File: pkg/server/container_create_unix_test.go [hunk 3]
```diff
@@ -1193,8 +1194,15 @@ func TestPrivilegedDevices(t *testing.T) {
 		spec, err := c.containerSpec(t.Name(), testSandboxID, testPid, "", testContainerName, containerConfig, sandboxConfig, imageConfig, nil, ociRuntime)
 		assert.NoError(t, err)
 
-		hostDevices, err := devices.HostDevices()
+		hostDevicesRaw, err := devices.HostDevices()
 		assert.NoError(t, err)
+		var hostDevices []*libcontainerconfigs.Device
+		for _, dev := range hostDevicesRaw {
+			// https://github.com/containerd/cri/pull/1521#issuecomment-652807951
```
Comment (mikebrow) [inline pkg/server/container_create_unix_test.go:1201]: nice!

File: vendor.conf (hunk 1 above)
Comment (AkihiroSuda) [reply vendor.conf:40]: errno of what?

File: vendor.conf (hunk 2 above)
Comment (AkihiroSuda) [reply vendor.conf:26]: ?

Comment (mikebrow) [reply vendor.conf:26]: normally we pull containerd dependencies when we update the version of containerd in vendor.  This case I see dims has the PR that is making these same vendor changes... So this is ok! Just making a comment while reviewing.

File: vendor.conf (hunk 1 above)
Comment (mikebrow) [reply vendor.conf:40]: https://github.com/opencontainers/runtime-spec/commit/023044323086583e5b0071ed1e6c18fd499836f3#diff-7f24d60f0cbb9c433e165467e3d34838R672

**Comment by dims** [comment]
LGTM we need this in https://github.com/containerd/containerd/pull/4357

File: vendor.conf [hunk 4]
```diff
@@ -36,28 +36,30 @@ github.com/Microsoft/go-winio                       v0.4.14
...
 golang.org/x/sync                                   42b317875d0fa942474b76e1b46a6060d720ae6e
-golang.org/x/sys                                    5c8b2ff67527cb88b770f693cebf3799036d8bc0
+golang.org/x/sys                                    9dae0f8f577553e0f21298e18926efc9644c281d
 golang.org/x/text                                   19e51611da83d6be54ddafce4a4af510cb3e9ea4
 google.golang.org/genproto                          e50cd9704f63023d62cd06a1994b98227fc4d21a
 google.golang.org/grpc                              v1.27.1
 
 # cgroups dependencies
-github.com/cilium/ebpf                              4032b1d8aae306b7bb94a2a11002932caf88c644
+github.com/cilium/ebpf                              a9f01edf17e335304d9bd17f40da4289442477c8
```
Comment (thaJeztah) [inline vendor.conf:62]: Can you update this to master to get rid of the golang.org/x/xerrors dependency?

File: vendor.conf [hunk 5]
```diff
@@ -36,28 +36,30 @@ github.com/Microsoft/go-winio                       v0.4.14
...
 github.com/prometheus/client_model                  v0.2.0
 github.com/prometheus/common                        v0.9.1
 github.com/prometheus/procfs                        v0.0.11
-github.com/russross/blackfriday                     v1.5.2
+github.com/russross/blackfriday/v2                  v2.0.1
+github.com/shurcooL/sanitized_anchor_name           v1.0.0
 github.com/sirupsen/logrus                          v1.6.0
 github.com/syndtr/gocapability                      d98352740cb2c55f81556b63d4a1ec64c5a319c2
-github.com/urfave/cli                               v1.22.0
+github.com/urfave/cli                               v1.22.1
```
Comment (thaJeztah) [inline vendor.conf:50]: Should we add the comment here as well?

File: vendor.conf (hunk 1 above)
Comment (thaJeztah) [inline vendor.conf:40]: Perhaps leave a `git describe -tags` comment to give some indication what version this is

**Comment by mikebrow** [comment]
Akihiro may be asleep.. we could just merge this and push another pr to make the additional updates...

**Comment by AkihiroSuda** [comment]
updated

**Pull request merged**